### ISSUE TITLE
A "hideLabel" prop for Dropdowns (specifically inline Dropdowns)

### ISSUE THREAD

**Issue by molyholy** [carbon-design-system/carbon#7571]
## Environment

> Browser
Firefox

> Assistive technology used to verify
IBM Equal Access Accessibility Checker

## Detailed description

> What WCAG 2.1 checkpoint does the issue violate?
IBM CI 162 Checkpoint Violation: 1.3.1 Info and Relationships
https://www.ibm.com/able/guidelines/ci162/info_and_relationships.html
https://www.ibm.com/able/guidelines/ci162/name_role_value.html

## Steps to reproduce the issue

1. Log into https://test.cloud.ibm.com/billing-nouveau/invoices
2. Run the Accessibility Checker

## Additional information

- Screenshots or code
<img width="1790" alt="image" src="https://user-images.githubusercontent.com/23204596/104644569-e8b3dd80-567b-11eb-83d3-2d79438ae006.png">

- Notes
- The inline Dropdown in the top right does not have a "Label" component, which creates an a11y issue. However, adding a label seems to defeat the purpose of the inline Dropdown (and if nothing else, looks kind of weird). The Select component already has a hideLabel prop which hides the label visually but not programatically, and I think the inline Dropdown could really benefit from something similar.

@dakahn @ryanomackey

**Comment by dakahn** [comment]
I'm cool with this. We have these on various Textinputs and the Select component.
